webdriver/tests/interaction/element_clear.py

```diff
@@ -0,0 +1,133 @@
+from tests.support.asserts import assert_error, assert_success
+from tests.support.inline import inline
+
+
+def clear(session, element):
+    return session.transport.send("POST", "session/{session_id}/element/{element_id}/clear"
+                                  .format(session_id=session.session_id,
+                                          element_id=element.id))
+
+
+# 14.2 Element Clear
+
+def test_no_browsing_context(session, create_window):
+    # 14.2 step 1
+    session.url = inline("<p>This is not an editable paragraph.")
+    element = session.find.css("p", all=False)
+
+    session.window_handle = create_window()
+    session.close()
+
+    response = clear(session, element)
+    assert_error(response, "no such window")
+
+
+def test_element_not_found(session):
+    # 14.2 Step 2
+    response = session.transport.send("POST", "session/{session_id}/element/{element_id}/clear"
+                                      .format(session_id=session.session_id,
+                                              element_id="box1"))
+
+    assert_error(response, "no such element")
+
+
+def test_element_not_editable(session):
+    # 14.2 Step 3
+    session.url = inline("<p>This is not an editable paragraph.")
+
+    element = session.find.css("p", all=False)
+    response = clear(session, element)
+    assert_error(response, "invalid element state")
+
+
+def test_element_not_resettable(session):
+    # 14.2 Step 3
+    session.url = inline("<input type=button value=Federer>")
```

Comment:
You should parameterize this test for all the [non-resettable elements](https://html.spec.whatwg.org/#category-reset).

Reply:
Hey @andreastt, 

I'm not 100% sure if I understand your comment. Would you like me to have an argument say "element" that could be an html tag that is a non-resettable element?

Also, to test all the non-resettable elements, do you want me to test all the elements that are not input, output, select, and textarea?

Reply:
Sorry if I was imprecise in my last comment. I would [parameterise](https://docs.pytest.org/en/latest/parametrize.html#pytest-mark-parametrize) this test so that `<input>`, `<output>`, `<select>`, and `<textarea>` are tested as resettable elements. According to HTML I can’t see that `<input type=button>` is a non-resettable element?

Reply:
<input>, <output>, <select>, and <textarea> are resettable elements. This test is suppose to test that if you try to clear a resettable element, it should throw the invalid element state error. The test should test if it throws the error correct?

And testing if <input>, <output>, <select>, and <textarea> should be in 14.2 Step 8?

Reply:
That sounds about right.